pkg/component/logging/vali/vali.go

```diff
@@ -0,0 +1,927 @@
+					Requests: corev1.ResourceList{
+						corev1.ResourceCPU:    resource.MustParse("5m"),
+						corev1.ResourceMemory: resource.MustParse("35Mi"),
+					},
+					Limits: corev1.ResourceList{
+						corev1.ResourceMemory: resource.MustParse("350Mi"),
+					},
+				},
+				SecurityContext: &corev1.SecurityContext{
+					Capabilities: &corev1.Capabilities{
+						Add: []corev1.Capability{"NET_ADMIN"},
+					},
+				},
+				Ports: []corev1.ContainerPort{{
+					Name:          telegrafName,
+					ContainerPort: telegrafServicePort,
+					Protocol:      corev1.ProtocolTCP,
+				}},
+				VolumeMounts: []corev1.VolumeMount{
+					{
+						Name:      telegrafVolumeName,
+						MountPath: telegrafVolumeMountPath + "/" + telegrafDataKeyConfig,
+						SubPath:   telegrafDataKeyConfig,
+						ReadOnly:  true,
+					},
+					{
+						Name:      telegrafVolumeName,
+						MountPath: telegrafVolumeMountPath + "/" + telegrafDataKeyStartScript,
+						SubPath:   telegrafDataKeyStartScript,
+						ReadOnly:  true,
+					},
+				},
+			},
+		)
+		statefulSet.Spec.Template.Spec.Volumes = append(statefulSet.Spec.Template.Spec.Volumes, corev1.Volume{
+			Name: telegrafVolumeName,
+			VolumeSource: corev1.VolumeSource{
+				ConfigMap: &corev1.ConfigMapVolumeSource{
+					LocalObjectReference: corev1.LocalObjectReference{
+						Name: telegrafConfigMapName,
+					},
+				},
+			},
+		})
+
+		// TODO: Move `pkg/component/logging/kuberbacproxy` resources into this component.
+		utilruntime.Must(gardenerutils.InjectGenericKubeconfig(statefulSet, genericTokenKubeconfigSecretName, "shoot-access-"+kubeRBACProxyName))
+	}
+
+	utilruntime.Must(references.InjectAnnotations(statefulSet))
+	return statefulSet
+}
+
+func getLabels() map[string]string {
+	return map[string]string{
+		v1beta1constants.GardenRole: v1beta1constants.GardenRoleLogging,
+		v1beta1constants.LabelRole:  "logging",
+		v1beta1constants.LabelApp:   valiName,
+	}
+}
+
+// resizeOrDeleteValiDataVolumeIfStorageNotTheSame updates the Vali PVC if passed storage value is not the same as the
+// current one.
+// Caution: If the passed storage capacity is less than the current one the existing PVC and its PV will be deleted.
+func (v *vali) resizeOrDeleteValiDataVolumeIfStorageNotTheSame(ctx context.Context) error {
+	managedResource := &resourcesv1alpha1.ManagedResource{ObjectMeta: metav1.ObjectMeta{Name: ManagedResourceControlName, Namespace: v.namespace}}
+	removeIgnoreAnnotationFromManagedResource := func() error {
+		// In order to not create the managed resource here first check if exists.
+		if err := v.client.Get(ctx, client.ObjectKeyFromObject(managedResource), managedResource); err != nil {
+			if !apierrors.IsNotFound(err) {
+				return err
+			}
+			return nil
+		}
+		_, err := controllerutils.GetAndCreateOrMergePatch(ctx, v.client, managedResource, func() error {
+			delete(managedResource.Annotations, resourcesv1alpha1.Ignore)
+			return nil
+		})
+		return err
```

Comment:
Now that we `Get` it upfront:

```suggestion
		patch := client.MergeFrom(managedResource.DeepCopy())
		delete(managedResource.Annotations, resourcesv1alpha1.Ignore)
		return v.client.Patch(ctx, managedResource, patch)
```